### PR TITLE
Account payment history

### PR DESCRIPTION
**Pull request by aonomike**
Added PaymentHistory.java
Added PaymentType.java
Added AccountRepository.java
Added PaymentHistoryRepository.java
Added AccountService.java
Deleted DripUser.java

### CODE REVIEW

**Comment by geoff-wasilwa** [comment]
could you squash your commits to one.

File: src/main/java/org/drip/model/PaymentType.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	private int id;
+	
+	@Column(name="payment_type_code")
+	private String paymentTypeCode;
+	
+	@Column(name="payment_type_description")
+	private String paymentTypeDescription;
+	
+	@Column(name="payment_type_sort")
+	private int paymentTypeSort;
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/model/PaymentType.java]: This field is not required

File: src/main/java/org/drip/model/PaymentType.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name="payment")
+public class PaymentType {
+	@Id 
+	@GeneratedValue(strategy=GenerationType.TABLE)
+	private int id;
+	
+	@Column(name="payment_type_code")
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/model/PaymentType.java]: change this to `code`.

File: src/main/java/org/drip/repository/AccountRepository.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.drip.model.Account;
+import org.drip.model.PaymentHistory;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface AccountRepository extends CrudRepository<Account, Long> {
+	@Query("SELECT paymentHistory FROM PaymentHistory paymentHistory INNER JOIN paymentHistory.account INNER JOIN paymentHistory.paymentType WHERE account.accountNumber=:accountNumber " )
+	PaymentHistory findPaymentHistoryByAccountNumber(@Param("accountNumber") String accountId);
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/repository/AccountRepository.java]: Remove this method from this class.

File: src/main/java/org/drip/repository/PaymentHistoryRepository.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.drip.model.PaymentHistory;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface PaymentHistoryRepository extends CrudRepository<PaymentHistory,Long> {
+
+	@Query("SELECT paymentHistory FROM PaymentHistory paymentHistory INNER JOIN paymentHistory.account INNER JOIN paymentHistory.paymentType WHERE account.accountNumber=:accountNumber " )
+	PaymentHistory findPaymentHistoryByAccountNumber(@Param("accountNumber") String accountId);
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/repository/PaymentHistoryRepository.java]: rename this to `findByAccountNumber`

File: src/main/java/org/drip/services/AccountService.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package org.drip.services;
+
+import org.drip.model.Account;
+import org.drip.model.PaymentHistory;
+
+public interface AccountService {
+	Account GetAccount(int customerId);
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/services/AccountService.java]: `getAccount`

File: src/main/java/org/drip/services/AccountService.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+	PaymentHistory GetPaymentHistory(String accountId);
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/services/AccountService.java]: `getPayment`

File: src/main/java/org/drip/repository/AccountRepository.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+	
+	@Query("SELECT account FROM Account account WHERE account.dripUser.id =:id")
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/repository/AccountRepository.java]: Rename `dripUser` to `Customer`

File: src/main/java/org/drip/services/impl/AccountServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+	
+	@Override
+	public Account GetAccount(int customerId) {
+		Account account = accountRepository.findByCustomerId(customerId);	
+		return account;
+	}
+
+	@Override
+	public PaymentHistory GetPaymentHistory(String accountId) {
+		// TODO Auto-generated method stub
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/services/impl/AccountServiceImpl.java]: delete this line

File: src/main/java/org/drip/model/PaymentHistory.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name="payment_history")
+public class PaymentHistory {
+	@Id
+	@GeneratedValue(strategy=GenerationType.TABLE)
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/model/PaymentHistory.java]: Change the generation type to `IDENTITY`

File: src/main/java/org/drip/model/Account.java [hunk 10]
```diff
@@ -1,50 +1,65 @@
...
+		this.customer = customer;
+	}
 	
-	public void setDripUser(Customer dripUser) {
-		this.customer = dripUser;
+	public List<Payment> getPayment() {
+		return payment;
+	}
+
+	public void setPayment(List<Payment> payment) {
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/model/Account.java]: remove set for `List` fields

File: src/main/java/org/drip/model/Payment.java [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+	@Column (name="payment_date")
+	private Date paymentDate;
+	
+	@Column(name="amount")
+	private Double amount;
+	
+	@Column(name="update_date")
+	private Date updateDate;	
+	
+	@ManyToOne(fetch=FetchType.LAZY)
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/model/Payment.java]: No need to specify `fetch`, the default value is `Lazy`

File: src/main/java/org/drip/repository/AccountRepository.java [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+import org.drip.model.Payment;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface AccountRepository extends CrudRepository<Account, Long> {
+	@Query("SELECT paymentHistory FROM PaymentHistory paymentHistory INNER JOIN paymentHistory.account INNER JOIN paymentHistory.paymentType WHERE account.accountNumber=:accountNumber " )
+	Payment findPaymentHistoryByAccountNumber(@Param("accountNumber") String accountId);
+	
+	@Query("SELECT account FROM Account account WHERE account.dripUser.id =:id")
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/repository/AccountRepository.java]: `dripUser` does not exists. Change this to `customer`

File: src/main/java/org/drip/repository/AccountRepository.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+
+import org.drip.model.Account;
+import org.drip.model.Payment;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface AccountRepository extends CrudRepository<Account, Long> {
+	@Query("SELECT paymentHistory FROM PaymentHistory paymentHistory INNER JOIN paymentHistory.account INNER JOIN paymentHistory.paymentType WHERE account.accountNumber=:accountNumber " )
+	Payment findPaymentHistoryByAccountNumber(@Param("accountNumber") String accountId);
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/repository/AccountRepository.java]: Remove this method, it should be in `PaymentRepository`

File: src/main/java/org/drip/repository/PaymentHistoryRepository.java [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+
+import org.drip.model.Payment;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface PaymentHistoryRepository extends CrudRepository<Payment,Long> {
+
+	@Query("SELECT paymentHistory FROM PaymentHistory paymentHistory INNER JOIN paymentHistory.account INNER JOIN paymentHistory.paymentType WHERE account.accountNumber=:accountNumber " )
+	Payment findPaymentHistoryByAccountNumber(@Param("accountNumber") String accountId);
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/repository/PaymentHistoryRepository.java]: change this to `findByAccountNumber`

File: src/main/java/org/drip/services/AccountService.java [hunk 15]
```diff
@@ -0,0 +1,10 @@
+package org.drip.services;
+
+import org.drip.model.Account;
+import org.drip.model.Payment;
+
+public interface AccountService {
+	Account GetAccount(int customerId);
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/services/AccountService.java]: Java methods should be in `camelCase`, first letter is NOT capitalized.

File: src/main/java/org/drip/repository/PaymentHistoryRepository.java [hunk 16]
```diff
@@ -0,0 +1,12 @@
+package org.drip.repository;
+
+import org.drip.model.Payment;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+public interface PaymentHistoryRepository extends CrudRepository<Payment,Long> {
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/repository/PaymentHistoryRepository.java]: Rename to `PaymentRepository`

File: src/main/java/org/drip/WebUserValidator.java [hunk 17]
```diff
@@ -56,6 +56,7 @@ public void validate(Object obj, Errors errors) {
 				errors.rejectValue("email", "email.unique");
 			}
 		}
+		//check if account name is empty
```
Comment (geoff-wasilwa) [inline src/main/java/org/drip/WebUserValidator.java]: Delete this comment

Comment (aonomike) [inline src/main/java/org/drip/WebUserValidator.java]: did this already, lemme check again...maybe am commiting the wrong way...leme check it out

**Pull request closed without merging**